app/assets/javascripts/host_edit_interfaces.js

```diff
@@ -391,9 +391,7 @@ $(document).on('change', '.virtual', function() {
 function construct_host_name() {
   var host_name_el = $('#host_name')
   var host_name = host_name_el.val();
-  if (host_name_el.data('appendDomainNameForHosts') === false ||
-      host_name_el.data('managed') === false
```

Comment:
It would be better to remove https://github.com/ofedoren/foreman/blob/develop/app/views/hosts/_form.html.erb#L36 as well.